### PR TITLE
feat: upload files with workers

### PR DESCRIPTION
**Pull request by Pierre-Do**
## Status
**READY**

## Description

The goal if this PR is to add a basic worker support to upload sourcemaps.

This worker support is needed on very large projects so all sourcemaps are not being uploaded in parallel (see https://github.com/honeybadger-io/honeybadger-webpack/issues/299) This should help with Dockerized environments, such as CIs (Jenkins, GitHub Actions, etc...)

The implementation revolves around two core tenets:

1. The promise created by `uploadSourceMap` is now wrapped in a closure, giving us more control on when the promise is created and thus the sourcemap file is uploaded
2. The promises are resolved by a generator; this generator can be consumed by a configurable number of workers

This implementation is based on this sandbox: https://github.com/Pierre-Do/promise-scheduler. The generator approach is shamelessly inspired from a publication by [Alex Ewerlöf](https://codeburst.io/async-map-with-limited-parallelism-in-node-js-2b91bd47af70)

This feature has been QA-ed on production environments using this fork: https://github.com/qoqa/honeybadger-webpack/commit/0078ee289937ea3aa37a437216cf072749a72e80. Cheers to my team at [QoQa](https://github.com/qoqa) for the support :100: 

Fix for https://github.com/honeybadger-io/honeybadger-webpack/issues/299

## Todos
- [x] Tests
- [x] Documentation
- [x] Changelog Entry (unreleased)

## Steps to Test or Reproduce

You can run all tests with the usual test command `npm test`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "clean": "rimraf dist",
     "prebuild": "npm run -s clean",
     "prepublishOnly": "npm run clean && npm run build && npm run test",
-    "test": "mocha --require @babel/register",
```
Comment (Pierre-Do) [inline package.json:12]: Small unrelated change: I've moved the `@babel/register` to a Mocha configuration file: this helps greatly with Webstorm default integration!

image

Comment (joshuap) [reply package.json:12]: Thanks!

**Review by joshuap — APPROVED** [review]
This is awesome. Thanks!

**Comment by ParamagicDev** [comment]
Woah awesome stuff! Thanks @Pierre-Do !

**Comment by Pierre-Do** [comment]
Thanks for your feedback @ParamagicDev and @joshuap, always happy to help.

Would you mind releasing a beta with this change?
